File: python/src/greeter.py
"""Greeter module providing localized greeting messages."""


SUPPORTED_LANGUAGES = ("en", "es", "fr", "de")


class Greeter:
    """Produces greeting strings for a given language."""

    def __init__(self, language: str = "en") -> None:
        self.language = language if language in SUPPORTED_LANGUAGES else "en"

    def greet(self, name: str) -> str:
        """Return a greeting for *name* in the configured language."""
        greetings = {
            "es": f"¡Hola, {name}!",
            "fr": f"Bonjour, {name}!",
            "de": f"Hallo, {name}!",
        }
        return greetings.get(self.language, f"Hello, {name}!")

File: cpp/src/greeter.cpp
#include "greeter.h"

Greeter::Greeter(const std::string& language) : language_(language) {}

std::string Greeter::greet(const std::string& name) const {
    if (language_ == "es") {
        return "¡Hola, " + name + "!";
    } else if (language_ == "fr") {
        return "Bonjour, " + name + "!";
    } else if (language_ == "de") {
        return "Hallo, " + name + "!";
    } else {
        return "Hello, " + name + "!";
    }
}

std::string Greeter::getLanguage() const {
    return language_;
}
